src/client/feature_proxy.rs

```diff
@@ -38,29 +38,15 @@ impl<'a> FeatureProxy<'a> {
 }
 
 impl<'a> Feature for FeatureProxy<'a> {
-    fn get_id(&self) -> &str {
-        &self.feature_id
-    }
-
     fn get_name(&self) -> crate::errors::Result<String> {
         self.client.get_feature(&self.feature_id)?.get_name()
     }
 
-    fn get_data_type(&self) -> crate::errors::Result<crate::models::ValueKind> {
-        self.client.get_feature(&self.feature_id)?.get_data_type()
-    }
-
     fn is_enabled(&self) -> crate::errors::Result<bool> {
         self.client.get_feature(&self.feature_id)?.is_enabled()
     }
```

Comment:
for the future:

Not sure if `is_enabled` this is something we want to expose to the user. The concept of "enabled" is hard to understand and does not consider entity evaluation.
It might be confusing to see the feature as `enabled` but when getting the value (using entity) a user will see the `disabled` value, as then rollout percentage and segment evaluation is considered.
It might tempt users to use the `is_enabled()` function to evaluate the feature (assuming most features are boolean) which would give wrong results.

maybe @saikumar1607 can help to clarify?

Reply:
If a feature is disabled, the user can choose not to evaluate it for the entities, but it doesn't mean that the user knows the enabled and disabled values.

> It might tempt users to use the is_enabled() function to evaluate the feature (assuming most features are boolean) which would give wrong results.

Only using our `get_value`, the user can retrieve the values.... and it requires the `Entity`:

```rust
fn get_value(&self, entity: &impl Entity) -> Result<Value>;
```

And of course, even if the features are boolean and "disabled" means "false", the user won't benefit of the rollout related capabilities provided by IBM AppConfiguration.

---

In my mind, knowing if a feature is enabled or not, could be needed for applications that might choose whether to load a module or not. For example, given an e-commerce, if the "groceries feature" is not enabled, I won't load that module (maybe at startup time). Or, for example, if the "advanced-pricing" is disabled, I will just use default pricing for all the users instead of running some evaluation for each product for each user.